From: David Bauer <mail@david-bauer.net>
Date: Mon, 27 Nov 2017 04:19:38 +0100
Subject: ar71xx: C58/C59 fix LAN1 working incorrectly

This commit fixes LAN Port 1 not transferring data in case no
other LAN Port has active link-state on TP-Link Archer C58/C59.

Signed-off-by: David Bauer <mail@david-bauer.net>

diff --git a/target/linux/ar71xx/files/arch/mips/ath79/mach-archer-c59-v1.c b/target/linux/ar71xx/files/arch/mips/ath79/mach-archer-c59-v1.c
index f385d4a5a3148b83ee01007145e0eda2c0ef670f..129aa53f304dd1a118ace9a2749855cb36f66cfc 100644
--- a/target/linux/ar71xx/files/arch/mips/ath79/mach-archer-c59-v1.c
+++ b/target/linux/ar71xx/files/arch/mips/ath79/mach-archer-c59-v1.c
@@ -249,7 +249,7 @@ static void __init archer_c5x_v1_setup(void)
 	ath79_eth1_data.phy_if_mode = PHY_INTERFACE_MODE_GMII;
 	ath79_eth1_data.speed = SPEED_1000;
 	ath79_eth1_data.duplex = DUPLEX_FULL;
-	ath79_switch_data.phy_poll_mask |= BIT(4);
+	ath79_switch_data.phy_poll_mask |= BIT(0);
 	ath79_switch_data.phy4_mii_en = 1;
 	ath79_register_eth(1);
 
